Crash on docker event handler

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x58390c]

goroutine 46 [running]:
github.com/weaveworks/scope/probe/docker.(*registry).handleEvent(0xc820307d00, 0x0)
        /go/src/github.com/weaveworks/scope/probe/docker/registry.go:230 +0x25c
github.com/weaveworks/scope/probe/docker.(*registry).listenForEvents(0xc820307d00, 0x0)
        /go/src/github.com/weaveworks/scope/probe/docker/registry.go:165 +0x4f1
github.com/weaveworks/scope/probe/docker.(*registry).loop(0xc820307d00)
        /go/src/github.com/weaveworks/scope/probe/docker/registry.go:121 +0x21
created by github.com/weaveworks/scope/probe/docker.NewRegistry
        /go/src/github.com/weaveworks/scope/probe/docker/registry.go:97 +0x282
```

Full logs: https://weaveworks.slack.com/files/ilya/F0ML19W9W/scope-1455661827.log.xz

Looks like we are receiving a nil event, probably the  docker client is closing the channel, which we are not handling gracefuly: https://github.com/weaveworks/scope/blob/v0.12.0/probe/docker/registry.go#L230

--- a/probe/docker/registry.go
+++ b/probe/docker/registry.go
@@ -161,7 +161,11 @@ func (r *registry) listenForEvents() bool {
 	otherUpdates := time.Tick(r.interval)
 	for {
 		select {
-		case event := <-events:
+		case event, ok := <-events:
+			if !ok {
+				log.Errorf("docker registry: event listener unexpectedly disconnected")
+				return true
+			}
 			r.handleEvent(event)
 
 		case <-otherUpdates:

